src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +27,24 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Rectangle rectangle = (Rectangle) o;
+        return width.equals(rectangle.width)
+                && length.equals(rectangle.length)
+                && color.equals(rectangle.color);
+    }
+
+    @Override
+    public int hashCode() {
+        return 31 * ((width != null) ? width.hashCode() : 0)
+                + 31 * ((length != null) ? length.hashCode() : 0) + color.length();
```

Comment:
color is also a field of Rectangle class. so if you ask its method without null check, you may get NullPointerException

Reply:
I had thought about it:(. Thank you!